src/config.js

```diff
@@ -395,6 +395,10 @@ export default class Config {
       return pkg.location;
     }
 
+    if (pkg.remote.type === 'copy') {
```

Comment:
The couple `pkg.remote.type === 'copy'` checks feel like an implementation detail that should be left to the `remote`. (leaky abstraction?) Perhaps this module path creation should be part of the `remote` and not part of `config` (config feels like a weird place for it anyway).

Reply:
Totally agree. I think we need to extend the `remote` information to allow "directly available" packages. This may become useful for other things in the future but I don't know how directly. Right now it feels specific to `link:` and `file:` protocols so I was unsure about extending this data structure or not.

Reply:
`link-resolver` has a type of `link` not `copy`, but `tarball-resolver` also has a type of `copy` so will also be affected by this change. Not sure if that is OK or not...

Reply:
Looks like `tarball-resolver` can also report a type of `copy`. Not sure if that is OK or not...

```
    // set remote so it can be "fetched"
    pkgJson._remote = {
      type: 'copy',
      resolved: `${url}#${hash}`,
      hash,
      registry,
      reference: dest,
    };
```